pandas/core/indexing.py

```diff
@@ -1550,6 +1551,13 @@ def _setitem_with_indexer(self, indexer, value):
                 val = list(value.values()) if isinstance(value, dict) else value
                 blk = self.obj._mgr.blocks[0]
                 take_split_path = not blk._can_hold_element(val)
```

Comment:
this can be the else condtiion

Reply:
No, value can be anything from int, float to numpy array. I think this check is only necessary if we have Series or DataFrame. Maybe with an array?